This PR fixes the following warnings reported in https://github.com/apache/hadoop/pull/2792 in CI run - https://github.com/apache/hadoop/pull/2792#issuecomment-808709573

```
[WARNING] /home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-2792/src/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/lib/common/hdfs_configuration.cc:138:22: warning: catching polymorphic type 'const class hdfs::uri_parse_error' by value [-Wcatch-value=]
[WARNING] /home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-2792/src/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/lib/common/hdfs_configuration.cc:151:27: warning: catching polymorphic type 'struct hdfs::ha_parse_error' by value [-Wcatch-value=]
```

--- a/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/lib/common/hdfs_configuration.cc
+++ b/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/lib/common/hdfs_configuration.cc
@@ -135,7 +135,7 @@ std::vector<NamenodeInfo> HdfsConfiguration::LookupNameService(const std::string
       URI uri;
       try {
         uri = URI::parse_from_string(PrependHdfsScheme(Get(dom_node_name)));
-      } catch (const uri_parse_error) {
+      } catch (const uri_parse_error&) {
         throw ha_parse_error("unable to find " + dom_node_name);
       }
 
@@ -148,7 +148,7 @@ std::vector<NamenodeInfo> HdfsConfiguration::LookupNameService(const std::string
       NamenodeInfo node(nameservice, *node_id, uri);
       namenodes.push_back(node);
     }
-  } catch (ha_parse_error e) {
+  } catch (const ha_parse_error& e) {
     LOG_ERROR(kRPC, << "HA cluster detected but failed because : " << e.what());
     namenodes.clear(); // Don't return inconsistent view
   }

